framework/validators/CompareValidator.php

```diff
@@ -151,6 +151,17 @@ public function validateAttribute($model, $attribute)
             $compareAttribute = $this->compareAttribute === null ? $attribute . '_repeat' : $this->compareAttribute;
             $compareValue = $model->$compareAttribute;
             $compareLabel = $compareValueOrAttribute = $model->getAttributeLabel($compareAttribute);
+
+            if ($model->hasErrors($compareAttribute)) {
```

Comment:
I think we should check here also if skipOnError is not true.

Reply:
It will be checked before this method is called

```php
public function validateAttributes($model, $attributes = null)
{
    $attributes = $this->getValidationAttributes($attributes);

    foreach ($attributes as $attribute) {
        $skip = $this->skipOnError && $model->hasErrors($attribute)
            || $this->skipOnEmpty && $this->isEmpty($model->$attribute);
        if (!$skip) {
            if ($this->when === null || call_user_func($this->when, $model, $attribute)) {
                $this->validateAttribute($model, $attribute);
            }
        }
    }
}
```

Reply:
That one takes care of main validated attribute, I mean the `$compareAttribute` here.

Reply:
not sure that it's possible with the current validators architecture since there is no way to get an exact rule, for example, if there are four separate rules for the same attribute and two of them have **skipOnError**, and two don't

Reply:
It is, we are checking that in the context of the current rule only so if the current rule has the skip on error option it should be applicable here as well (IMO at least).

Reply:
do you mean that it should be like that?

```php
if ($model->hasErrors($compareAttribute) && $this->skipOnError !== true) {
  $this->addError(
      $model, 
      $attribute, 
      Yii::t('yii', '{compareAttribute} is invalid.'), 
      ['compareAttribute' => $compareLabel]
  );

  return;
}
```

Reply:
Yes.

Reply:
added this condition